dataprofiler/profilers/profile_builder.py

```diff
@@ -1355,11 +1362,118 @@ def _update_row_statistics(self, data, sample_ids=None):
             self.row_has_null_count = len(null_in_row_count)
             self.row_is_null_count = len(null_rows)
 
+    def _update_correlation(self, clean_samples):
+        """
+        Update correlation matrix for cleaned data.
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        columns = self.options.correlation.columns
+        if columns is None:
+            for col in self._profile:
+                if self._profile[col].profile['data_type'] not in ['int', 'float'] \
+                or self._profile[col].profile['statistics']['null_count'] > 0:
+                    clean_samples.pop(col)
+        if len(clean_samples) <= 1:
+            return
+
+        data = pd.DataFrame(clean_samples)
+        data = data.apply(pd.to_numeric, errors='coerce')
+        correlation_matrix = data.corr()
+
+        if self.correlation_matrix is None:
+            self.correlation_matrix = correlation_matrix
+            return
+
+        # currently return None with the exising correlation
+        # TODO: implement the update with the exising correlation
+        warnings.warn("Currently the update with existing correlation "
+                      "doesnt work.")
+        return
+
+    def _merge_correlation(self, other):
+        """
+        Merge correlation matrix from two profiles
+
+        :param other: the other profile that needs to be merged
+        :return:
+        """
+        corr_mat1 = self.correlation_matrix
+        mean1 = np.array(
+            [self._profile[profile_name].profile['statistics']['mean']
+             for profile_name in self._profile])
+        std1 = np.array(
+            [self._profile[profile_name].profile['statistics']['stddev']
+             for profile_name in self._profile])
+        n1 = self.total_samples
+
+        corr_mat2 = other.correlation_matrix
+        mean2 = np.array(
+            [other._profile[profile_name].profile['statistics']['mean']
+             for profile_name in other._profile])
+        std2 = np.array(
+            [other._profile[profile_name].profile['statistics']['stddev']
+             for profile_name in other._profile])
+        n2 = other.total_samples
+        return self._merge_correlation_helper(corr_mat1, mean1, std1, n1,
+                                              corr_mat2, mean2, std2, n2)
```

Comment:
I'm confused,  thought this doesn't work from our previous talks, does it work for merging two separate profiles?

Reply:
as we discussed, this merge works for two profiles without nan values. I think you mentioned about the `update_correlation`, not the merge.